### ISSUE TITLE
Postfix arg for MetricCollection.clone in order to support TensorBoard grouping

### ISSUE THREAD

**Issue by EduardKononov** [PyTorchLightning/metrics#168]
TensorBoard supports metrics grouping by name using forward slash. It's really convenient in most situations.
There are two ways of grouping:
* using prefix: "train/accuracy", "valid/accuracy"
* using postfix: "accuracy/train", "accuracy/valid" (by the way, this one is [demonstrated in the official PyTorch documentation](https://pytorch.org/docs/stable/tensorboard.html))
The first one can be easily implemented with [torchmetrics.MetricCollection.clone](https://github.com/PyTorchLightning/metrics/blob/master/torchmetrics/collections.py#L130) because it has `prefix` attribute:
```
metrics = MetricCollection(Accuracy(), Precision(), Recall())
train_metrics = metrics.clone(prefix='train/')
valid_metrics = metrics.clone(prefix='valid/')
```
But if one wants to use postfix grouping, there is no such a beautiful way.
What do you think of adding `postfix` arg?
```
metrics = MetricCollection(Accuracy(), Precision(), Recall())
train_metrics = metrics.clone(postfix='/train')
valid_metrics = metrics.clone(postfix='/valid')
```

**Comment by SkafteNicki** [comment]
@EduardKononov great enhancement! Want to send a PR? :]
